src/goto-programs/replace_function_bodies.cpp

```diff
@@ -0,0 +1,462 @@
+        std::regex_match(id2string(symbol.first), globals_regex))
+      {
+        if(!is_constant_or_has_constant_components(symbol.second.type, ns))
+        {
+          globals_to_havoc.push_back(symbol.first);
+        }
+        else if(!symbol.second.type.get_bool(ID_C_constant))
+        {
+          messages.warning() << "warning: not havoccing non-const global '"
+                             << symbol.first << "' because it has const members"
+                             << messaget::eom;
+        }
+      }
+    }
+    return util_make_unique<havoc_replace_function_bodiest>(
+      std::move(globals_to_havoc), std::move(params_regex), message_handler);
+  }
+  throw replace_function_bodies_errort("Can't parse \"" + options + "\"");
+}
+
+/// Produce a replace_function_body_optionst from a cmdline.
+/// \see replace_function_bodies() for information of how the cmdline is parsed
+replace_function_body_optionst parse_replace_function_body_optionst(
+  const cmdlinet &cmdline,
+  const symbol_tablet &symbol_table,
+  message_handlert &message_handler)
+{
+  if(
+    cmdline.isset("replace-function-body") &&
+    cmdline.isset("generate-function-body"))
+  {
+    throw replace_function_bodies_errort(
+      "can only use one of --replace-function-body "
+      "or --generate-function-body");
+  }
+  bool only_generate = cmdline.isset("generate-function-body");
+  std::regex function_regex = std::regex(
+    only_generate ? cmdline.get_value("generate-function-body")
+                  : cmdline.get_value("replace-function-body"));
+  auto replace_function_bodies = replace_function_body_options_factory(
+    cmdline.get_value("replace-function-body-options"),
+    symbol_table,
+    message_handler);
+  return {function_regex, only_generate, std::move(replace_function_bodies)};
+}
+
+replace_function_body_optionst::replace_function_body_optionst(
+  std::regex functions_regex,
+  bool only_generate,
+  std::unique_ptr<replace_function_bodiest> replace_function_bodies)
+  : functions_regex(std::move(functions_regex)),
+    only_generate(only_generate),
+    replace_function_bodies(std::move(replace_function_bodies))
+{
+}
+
+replace_function_body_optionst::replace_function_body_optionst(
+  replace_function_body_optionst &&other)
+  : functions_regex(std::move(other.functions_regex)),
+    only_generate(other.only_generate),
+    replace_function_bodies(std::move(other.replace_function_bodies))
+{
+}
+
+replace_function_body_optionst &replace_function_body_optionst::
+operator=(replace_function_body_optionst &&other)
+{
+  functions_regex = std::move(other.functions_regex);
+  only_generate = other.only_generate;
+  replace_function_bodies = std::move(other.replace_function_bodies);
+  return *this;
+}
+
+/// Replace function bodies with some default behavior
+/// A list of currently accepted command line arguments
+/// + the type of bodies generated by them
+///
+/// assert-false: { __CPROVER_assert(false); }
```

Comment:
`__CPROVER_assert` takes a second argument.

Reply:
Ah. I should probably just use assert here, the __CPROVER doesn't really add to the legibility of the example